packages/@react-aria/tooltip/test/useTooltip.test.js

```diff
@@ -10,18 +10,211 @@
 import React from 'react';
-import {renderHook} from '@testing-library/react-hooks';
-import {useTooltip} from '../';
-
-describe.skip('useTooltip', function () {
-  let renderTooltipHook = (props) => {
-    let {result} = renderHook(() => useTooltip(props));
-    return result.current;
-  };
-
-  it('handles defaults', function () {
-    let {tooltipProps} = renderTooltipHook({children: 'Test Tooltip'});
-    expect(tooltipProps.role).toBe('tooltip');
+import {useInteractionModality} from '@react-aria/interactions';
+import {useTooltip, useTooltipTrigger} from '../';
+import {useTooltipTriggerState} from '@react-stately/tooltip';
+
+describe('useTooltip', function () {
+  afterEach(() => {
+    jest.useRealTimers();
+  });
+  it('opens tooltip on hover after delay', function () {
+    jest.useFakeTimers('modern');
+
+    function Test() {
+      useInteractionModality();
+
+      const state = useTooltipTriggerState();
+      const {triggerProps, tooltipProps} = useTooltipTrigger({}, state);
+      const {tooltipProps: finalTooltipProps} = useTooltip(tooltipProps);
+
+      return (
+        <div>
+          <span {...triggerProps}>{'Trigger'}</span>
+          {state.isOpen && <span {...finalTooltipProps}>{'Tooltip'}</span>}
+        </div>
+      );
+    }
+
+    const {container, queryByRole} = render(<Test />);
+    const triggerElement = container.firstChild.firstChild;
+
+    // trigger pointer modality
+    fireEvent.mouseMove(container);
+
+    fireEvent.mouseEnter(triggerElement);
+
+    expect(queryByRole('tooltip')).toBeNull();
+    act(jest.runAllTimers);
+    expect(queryByRole('tooltip')).not.toBeNull();
+  });
+  it('opens tooltip immediately on hover with `delay: 0`', function () {
+    function Test() {
+      useInteractionModality();
+
+      const state = useTooltipTriggerState({delay: 0});
+      const {triggerProps, tooltipProps} = useTooltipTrigger({}, state);
+      const {tooltipProps: finalTooltipProps} = useTooltip(tooltipProps);
+
+      return (
+        <div>
+          <span {...triggerProps}>{'Trigger'}</span>
+          {state.isOpen && <span {...finalTooltipProps}>{'Tooltip'}</span>}
+        </div>
+      );
+    }
+
+    const {container, queryByRole} = render(<Test />);
+    const triggerElement = container.firstChild.firstChild;
+
+    // trigger pointer modality
+    fireEvent.mouseMove(container);
+
+    fireEvent.mouseEnter(triggerElement);
+
+    expect(queryByRole('tooltip')).not.toBeNull();
+  });
+
+  it('hides tooltip when hover leaves the trigger', function () {
+    jest.useFakeTimers('modern');
+
+    function Test() {
+      useInteractionModality();
+
+      const state = useTooltipTriggerState({delay: 0});
+      const {triggerProps, tooltipProps} = useTooltipTrigger({}, state);
+      const {tooltipProps: finalTooltipProps} = useTooltip(tooltipProps);
+
+      return (
+        <div>
+          <span {...triggerProps}>{'Trigger'}</span>
```

Comment:
```suggestion
          <span {...triggerProps}>Trigger</span>
```

Reply:
Changed all occurrences of this pattern in a file 👍